Snakefile

```diff
@@ -236,6 +243,19 @@ rule summary_table:
         summary_df = summary.summarize_networks(input.pathways, node_table)
         summary_df.to_csv(output.summary_table, sep='\t', index=False)
 
+rule ml: 
+    input: 
+        pathways = expand('{out_dir}{sep}{{dataset}}-{algorithm_params}{sep}pathway.txt', out_dir=out_dir, sep=SEP, algorithm_params=algorithms_with_params)
+    output: 
+        pca_image = SEP.join([out_dir, '{dataset}-{algorithm}-{params}', 'pca_image.png']),
```

Comment:
I believe the output file pattern will change so that `{algorithm}-{params}` is removed. The input pattern may be okay.